bitnami/keycloak/values.yaml

```diff
@@ -0,0 +1,635 @@
+  ##     path: /
+  ##
+  extraHosts: []
+
+  ## The tls configuration for additional hostnames to be covered with this ingress record.
+  ## see: https://kubernetes.io/docs/concepts/services-networking/ingress/#tls
+  ## Example:
+  ## extraTls:
+  ## - hosts:
+  ##     - keycloak.local
+  ##   secretName: keycloak.local-tls
+  ##
+  extraTls: []
+
+  ## If you're providing your own certificates, please use this to add the certificates as secrets
+  ## key and certificate should start with -----BEGIN CERTIFICATE----- or -----BEGIN RSA PRIVATE KEY-----
+  ## name should line up with a secretName set further up
+  ##
+  ## If it is not set and you're using cert-manager, this is unneeded, as it will create the secret for you
+  ## If it is not set and you're NOT using cert-manager either, self-signed certificates will be created
+  ## It is also possible to create and manage the certificates outside of this helm chart
+  ## Please see README.md for more information
+  ##
+  ## Example
+  ## secrets:
+  ##   - name: aspnet-core.local-tls
+  ##     key: ""
+  ##     certificate: ""
+  ##
+  secrets: []
+
+## Network Policy configuration
+## ref: https://kubernetes.io/docs/concepts/services-networking/network-policies/
+##
+networkPolicy:
+  ## Enable creation of NetworkPolicy resources
+  ##
+  enabled: false
+  ## The Policy model to apply. When set to false, only pods with the correct
+  ## client label will have network access to the ports Keycloak is listening
+  ## on. When true, Keycloak will accept connections from any source
+  ## (with the correct destination port).
+  ##
+  allowExternal: true
+  ## Additional NetworkPolicy Ingress "from" rules to set. Note that all rules are OR-ed.
+  ## Example:
+  ## additionalRules:
+  ##   - matchLabels:
+  ##       - role: frontend
+  ##   - matchExpressions:
+  ##       - key: role
+  ##         operator: In
+  ##         values:
+  ##           - frontend
+  ##
+  additionalRules: {}
+
+## Specifies whether RBAC resources should be created
+##
+rbac:
+  create: false
+  ## Custom RBAC rules
+  ## Example:
+  ## rules:
+  ##   - apiGroups:
+  ##       - ""
+  ##     resources:
+  ##       - pods
+  ##     verbs:
+  ##       - get
+  ##       - list
+  ##
+  rules: []
+
+## Specifies whether a ServiceAccount should be created
+##
+serviceAccount:
+  create: true
+  ## The name of the ServiceAccount to use.
+  ## If not set and create is true, a name is generated using the fullname template
+  ##
+  name: ""
+
+## Keycloak Pod Disruption Budget configuration
+## ref: https://kubernetes.io/docs/tasks/run-application/configure-pdb/
+##
+pdb:
+  create: false
+  ## Min number of pods that must still be available after the eviction
+  ##
+  minAvailable: 1
+  ## Max number of pods that can be unavailable after the eviction
+  ##
+  # maxUnavailable: 1
+
+## Keycloak Autoscaling configuration
+##
+autoscaling:
+  enabled: false
+  minReplicas: 1
+  maxReplicas: 11
+  # targetCPU: 50
+  # targetMemory: 50
```

Comment:
Can we standardize this object as https://github.com/bitnami/charts/blob/25a18ae364ac90383d4aa1d022fbc4c25af8403d/bitnami/airflow/values.yaml#L399?

```
autoscaling:
  enabled: false
  replicas:
    min: 1
    max: 11
  targets:
    cpu: 50
    memory: 50
```

Reply:
Why? That's not supported on Airflow anymore and other charts are using the format I used.

Reply:
The [PR](https://github.com/bitnami/charts/pull/4069) adds this feature to airflow. I was just a proposal, I like to see this

```
autoscaling.replicas.min=1,autoscaling.replicas.max=11
```

more than

```
autoscaling.minReplicas=1,autoscaling.maxReplicas=11
```

It could be just an OCD on my mind 😅. Do you think we can propose it as the new standard?

Reply:
I'd rather use `autoscaling.minReplicas`, `autoscaling.maxReplica` since it's more accurate to the HPA spec. See:

- https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#horizontalpodautoscalerspec-v1-autoscaling